### PR TITLE
Implement fontscaling

### PR DESCRIPTION
**Pull request by philippjfr**
Fixes https://github.com/holoviz/holoviews/issues/4107

Here's 1.4x fontscaling for bokeh and matplotlib:

Screen Shot 2020-01-06 at 2 25 56 PM

Screen Shot 2020-01-06 at 2 32 05 PM

### CODE REVIEW

File: holoviews/plotting/plot.py [hunk 1]
```diff
@@ -508,29 +512,76 @@ def _format_title_components(self, key, dimensions=True, separator='\n'):
         return (self.label, self.group, type(self).__name__, '')
 
 
+    def _get_fontsize_defaults(self):
+        """
+        Must returns default fontsize for the following objects:
```
Comment (jbednar) [inline holoviews/plotting/plot.py]:
```suggestion
        Must return default fontsize for the following objects:
```

**Comment by philippjfr** [comment]
>Could this at some point be part of an even higher-level property that acts like fig_dpi , scaling both fonts and all line, dot, and other glyph sizes to allow increasing the resolution while keeping the plot otherwise the same?

Yes, I like the idea, seaborn has something similar, it provides plotting contexts with various scaling factors, which scale font sizes, linewidths, mark sizes and tick sizes:

```
scaling = dict(paper=.8, notebook=1, talk=1.5, poster=2)
```

**Comment by jlstevens** [comment]
Looks good! I like the concept and I think this will be useful in the general case.

The only thing I am not entirely sure about is whether you should scale when an explicit override is specified. I think of scaling as a quick and easy way to get font sizes roughly where you want them without specifying absolute values. If absolute values are set, I feel those should be respected without applying font scaling. @jbednar @philippjfr Thoughts?

**Comment by philippjfr** [comment]
Just to record my thinking, I can see both arguments. The way I see it fontscaling is an option you might apply when exporting to different formats, e.g. for a talk or poster, and so it might make sense to scale the overrides too in that case, but I can definitely see that you might expect your override to be respected too. In the case of seaborn the fontscale option is global and modifies the defaults only, so I lean towards changing it to match that behavior but I'll let @jbednar chime in.

**Comment by jbednar** [comment]
Same here -- I can't decide which is more important: being able to specify an override that is always true regardless of overall size (e.g. to have a watermark or annotation that is always the same size in pixels regardless of scaling, like the map tile attributions in Bokeh), or to be able to confidently rescale the entire thing in one call, overrides and all.  I think I lean towards the latter (fontscale controls overrides as well) as being the least surprising, but would be happy with either behavior.

**Comment by jlstevens** [comment]
@jbednar As non of us seems to have a strong opinion, it sounds like we are leaving this up to you,  @philippjfr.

**Comment by philippjfr** [comment]
Okay, after thinking about it more I think scaling everything consistently is the least surprising behavior as well so I'll leave things as they are.

**Comment by SandervandenOord** [comment]
Just did some checks on plots I use.
```my_plot.opts(fontscale=1.5)``` works nicely for my plots :)

**Pull request merged**